core/pkg/store/flags.go

```diff
@@ -12,7 +12,7 @@ import (
 )
 
 type IStore interface {
-	GetAll(ctx context.Context) map[string]model.Flag
+	GetAll(ctx context.Context) (map[string]model.Flag, error)
```

Comment:
why this change? 🤔 it looks like only `nil` is returned

Reply:
@thisthat good question. The implementation of flagd currently does not use error return. But this change allows us to add an error if we see the need. But the main usage is for custom IStore implementations which use flagd's JSON evaluator [1]. These users now can use the evaluator and and return store-bound errors (ex:- Store access error if store implements a database access or another API)

[1] - https://github.com/open-feature/flagd/blob/main/core/pkg/evaluator/json.go#L73